evoseal/core/workflow.py

```
@@ -262,50 +343,192 @@ def _execute_step(self, step: dict[str, Any]) -> Any | None:
+                loop
+            )
+            # Don't wait for completion to avoid blocking
+            return
+        except RuntimeError:
+            # No running event loop, run synchronously in a new event loop
+            try:
+                loop = asyncio.get_event_loop()
+                if loop.is_running():
+                    # This should be rare, but handle the case where get_event_loop
+                    # returns a running loop after get_running_loop failed
+                    future = asyncio.run_coroutine_threadsafe(
```

Ruff (0.11.9)

Local variable `future` is assigned to but never used

Remove assignment to unused variable `future`

(F841)